configure.ac

```diff
@@ -3157,8 +3157,13 @@ then
   then
     AC_MSG_ERROR([use both --with-tcltk-includes='...' and --with-tcltk-libs='...' or neither])
   fi
-  TCLTK_INCLUDES=""
-  TCLTK_LIBS=""
+  if test -n "$PKG_CONFIG" && "$PKG_CONFIG" --exists tcl tk; then
+    TCLTK_INCLUDES="`"$PKG_CONFIG" tcl tk --cflags-only-I 2>/dev/null`"
+    TCLTK_LIBS="`"$PKG_CONFIG" tcl tk --libs 2>/dev/null`"
+  else
+    TCLTK_INCLUDES=""
+    TCLTK_LIBS=""
+  fi
```

Comment:
Could you please use autotools macros instead of manually running pkg-config?

https://autotools.io/pkgconfig/pkg_check_modules.html

Reply:
This would be inconsistent with the rest of `configure.ac`. I believe this change should be made in a separate PR for all the uses of pkg-config.

Reply:
@tiran I've checked in more detail about using `PKG_CHECK_MODULES` instead of the current approach.

I would be against using `PKG_CHECK_MODULES` for the following reasons:

1. Introduces an inconsistency with libffi which also runs pkg-config manually.
2. Introduces a naming inconsistency because of the `TCLTK_CFLAGS` variable that will be created. We already have `TCLTK_INCLUDES` which is essentially the same. Also, openssl uses `OPENSSL_INCLUDES`.
3. Introduces a naming conflict because of the `TCLTK_LIBS` variable.
4. Replaces ~10L of straightforward and arguably robust code (the libffi case hails from 2010) with >100L of autogenerated clutter.

To address 1-3, changes that go beyond the scope of this PR need to be introduced. If 1-3 are not addressed, the resulting configuration script will include inconsistencies that will have to be addressed before they get worse.

Given this, I believe it is better to go with manually running pkg-config for now. The use of `PKG_CHECK_MODULES` should be discussed in a separate issue, so the changes are applied in a uniform way so that there are no inconsistencies in the produced configure script.

Let me know what you think. Anyone else wishing to weigh-in, please let us know of your thoughts on this.

Reply:
I agree with @m000 here.  Tcl/Tk version selection has various legacy behaviors especially on macOS.  Trying to convert this to use a configure time check with an autotools macros would break compatibility.  I would be more in favor of using such an approach if and when we re-write the current Unix-y build system to address third-party library selection for all modules.

Reply:
No further updates from @tiran. Marking discussion as resolved since @ned-deily agrees with my points above.